### PR TITLE
Fix bug with indentation + soft indexing

### LINKED ISSUE #879: SUSHI 2.0.0 Beta 2: "The first index in a Soft Indexing sequence must be "+", an actual index of "0" has been assumed" although the first index is '+'
While using SUSHI 2.0.0 Beta 2 I receive the error "The first index in a Soft Indexing sequence must be "+", an actual index of "0" has been assumed" although the first index is designated as '+'. This happens if using soft indexing **on extensions** like this:

```
* extension[+].url = "testurl"
* extension[=].valueString = "{variable}"
```

SUSHI 2.0.0 Beta 1 and SUSHI stable run through just fine though!

### PR DESCRIPTION
**Pull request by ngfreiter**
Fixes #879, see that issue for further details.

This fixes a bug that happened when a soft indexed path was indented below another path, and the soft indexed path was _not_ a `PathRule`. See https://fshschool.org/FSHOnline/#/share/3sDl7Lm for a minimal example.

This PR fixes that by removing what appears to be a completely un-needed bit of code that was causing this bug. I believe that code was a relic from when I combined the handling of path context for normal rules and `CodeCaretRules`. But please as reviewers take a second to see if you could come up with a test case that would make the reviewed code necessary, because I would like that to get double checked.

### CODE REVIEW

**Review by jafeltra — COMMENTED** [review]
This looks good to me. I couldn't think of a case that isn't tested yet, and I ran a full regression and no projects had changes, so I think it's safe to remove that piece of code. I just had one minor suggestion.

**Review by mint-thompson — COMMENTED** [review]
This looks correct to me. The existing code that changes `[+]` to `[=]` for the topmost context after it is used, but before the new context is pushed onto the stack, guarantees that a `[+]` only gets used once. Because the stack grows one at a time, the topmost context is necessarily used in cases where the stack is growing. So, no matter how many levels we pop off the stack at a time, all previous contexts will have their `[+]` changed to `[=]`, which is what we want.

The only other case that still gets this error is one resulting from suspicious-looking FSH, where a context is used (and therefore has `[+]` become `[=]`), but only by path rules, which don't actually create rules on an entity. For example:
```
Instance: Foo
InstanceOf: PractitionerRole
* availableTime[+]
  * daysOfWeek[+]
  * allDay = true
```
The Instance will have just one rule on it, an AssignmentRule with a path of `availableTime[=].allDay`. However, I am comfortable with an error appearing in this case.

In conclusion: once you remove the comment that @jafeltra noted, this PR should be ready to go.

File: src/import/FSHImporter.ts [hunk 1]
```diff
@@ -2057,13 +2057,10 @@ export class FSHImporter extends FSHVisitor {
     }
 
     // Replace '[+]' with '[=]' in the version of the path to add to pathContext, unless the rule is only used to set a path
```
[the remark this answers is not in the document]
Comment (ngfreiter) [reply src/import/FSHImporter.ts]: Should be fixed by https://github.com/FHIR/sushi/pull/907/commits/60dec8510003dcf32a9ec0010e56f0e0704149bd. I also added a more explicit test since figured that may be a good idea.

**Review by jafeltra — APPROVED** [review]
Looks good and I like the bonus test!

**Review by mint-thompson — APPROVED** [review]
